Header for protected zone (#26)

aashapovalov opened the pull request:
<img width="1080" height="178" alt="Screenshot 2025-09-08 at 11 42 11" src="https://github.com/user-attachments/assets/cfa7c6a3-64ac-497b-a29d-91d7c426a1f8" />


# PR: Add Protected Zone Header Components

## Summary
- Added **MethodSwitch** component for HTTP method selection
- Implemented **LeftHeaderGroup** with method selector, input field, Send button, and tab navigation
- Implemented **RightHeaderGroup** with status, size, time indicators, and response label
- Added **StickyHeaderGrid** to combine left/right header groups in a unified layout 
- Updated **Protected Layout** to include sticky header grid:contentReference

## Notes
All components are styled with Tailwind and localized, but there is an error: "Could not resolve `tabs` in messages for locale `be`/`en`/`ru`", which I can't resolve. I'll try in the evening one more time, but if someone can look at it and help me debugging - it would be great.

## Important
Components are nested at `app/[locale]/public/layout.tsx` because we don't have protected zone yet. But component files are in `app/[locale]/public/_components` folder

aashapovalov committed:
feat: create element for switching HTTP methods(#26)

aashapovalov committed:
feat: create left header group and integrate method switch(#26)

aashapovalov committed:
feat: create right header group(#26)

aashapovalov committed:
feat: integrate left and right header in header grid and nested it in public layout(#26)

aashapovalov committed:
fix: change z-index of language switch to be visible on top of protected layout(#26)

dariadrozdova reviewed (changes requested):
What do you think about adding `cursor-pointer` to all clickable elements like tabs, method selector dropdown, `Send` button?

I noticed that the dropdown for choosing a language in the header does not work correctly in this branch. I think this needs to be fixed before the merge.

Also I think the dropdown with methods should close when click outside of it.

aashapovalov committed:
feat: add localization for right pane and send button(#26)

aashapovalov committed:
feat: add localization for left pane(#26)

aashapovalov committed:
feat: add hover effects for buttons(closes #26)

aashapovalov commented:
@dariadrozdova 
> What do you think about adding cursor-pointer to all clickable elements like tabs, method selector dropdown, Send button?

reasonable, I've added cursor-pointer
> I noticed that the dropdown for choosing a language in the header does not work correctly in this branch. I think this needs to be fixed before the merge.

what do you mean? I've experienced only issue that language switch element's z-index was less than protected-header's z-index, but I fixed it
> Also I think the dropdown with methods should close when click outside of it.

I agree, I wanted to develop custom hook to close these elements with `Esc` and outside click. I created a new task devoted to this issue.

aashapovalov committed:
feat: add cursor-pointer for active elements(closes #26)

dariadrozdova committed:
fix: update localization structure and improve translation handling

dariadrozdova committed:
Merge pull request #12 from dariadrozdova/fix/26_fix-localization-for-layout-protected-zone

Update localization structure and improve translation handling
